test-data/unit/check-custom-plugin.test

```diff
@@ -730,3 +730,15 @@ reveal_type(dynamic_signature(1))  # N: Revealed type is 'builtins.int'
 [file mypy.ini]
 \[mypy]
 plugins=<ROOT>/test-data/unit/plugins/function_sig_hook.py
+
+[case testClassAttrPluginFile]
+# flags: --config-file tmp/mypy.ini
+
+class Cls:
+    attr = 'test'
+
+reveal_type(Cls.attr)  # N: Revealed type is 'builtins.int'
+reveal_type(Cls().attr)  # N: Revealed type is 'builtins.str'
```

Comment:
Ideas for more test coverage:

* Test other exit points as well.
* Test using `x.attr` so that `x` has type `Type[Cls]`. 
* Test accessing another attribute that doesn't have a hook associated with.

Reply:
Ah, those are good ones! Would you recommend putting those in the same test case or lean more toward making separate test cases?

Reply:
It's fine to group related checks in a single test case (this makes tests run faster). These are probably fine in this test case, but if the test case gets pretty long (use your judgment) it may be better to split it.